types/compare.go

```diff
@@ -35,6 +37,50 @@ func CompareUint64(x, y uint64) int {
 	return 1
 }
 
+//VecComparUU returns []int64 comparing the []uint64 x to []uint64 y
+func VecComparUU(x, y []uint64, res []int64) {
+	n := len(x)
+	for i := 0; i < n; i++ {
+		if x[i] < y[i] {
+			res[i] = -1
+		} else if x[i] == y[i] {
+			res[i] = 0
+		} else {
+			res[i] = 1
+		}
+	}
+}
+
+//VecCompareII returns []int64 comparing the []int64 x to []int64 y
+func VecCompareII(x, y, res []int64) {
+	n := len(x)
+	for i := 0; i < n; i++ {
+		if x[i] < y[i] {
+			res[i] = -1
+		} else if x[i] == y[i] {
+			res[i] = 0
+		} else {
+			res[i] = 1
+		}
+	}
+}
+
+//VecComparUI returns []int64 comparing the []uint64 x to []int64y
+func VecComparUI(x []uint64, y, res []int64) {
```

Comment:
`VecComparUI(x []uint64, y, res []int64, n)`

Reply:
I had refined it. Thank you ~